util/cloud/s3.cc

```diff
@@ -165,8 +171,20 @@ std::error_code S3Bucket::Connect(uint32_t ms) {
   return http_client_->Connect(host, "80");
 }
 
-error_code S3Bucket::ListObjects(string_view path, ListObjectCb cb) {
-  h2::request<h2::empty_body> req{h2::verb::get, "/", 11};
+ListObjectsResult S3Bucket::ListObjects(string_view bucket_path, ListObjectCb cb,
+                                        std::string_view marker, int max_keys) {
+  std::string path = absl::StrCat("/", bucket_path, "?");
+
+  if (marker != "")
+    path += absl::StrCat("marker=", marker, "&");
+
+  if (max_keys != kAwsMaxKeys)
+    path += absl::StrCat("max-keys=", max_keys, "&");
+
+  if (path.back() == '?' || path.back() == '&')
+    path.pop_back();
```

Comment:
The If clause is always true. A better alternative would be:
```c++
CHECK(path.back() == '?' || path.back() == '&');
path.pop_back();
```

Reply:
i agree